src/weather_magic/core.cljs

```diff
@@ -47,23 +47,34 @@
                             :blend-fn [glc/src-alpha
                                        glc/one-minus-src-alpha]}))
 
-(defn update-year-month-info
-  [t key]
-  (let [min  (:min (:year (key @state/date-atom)))
-        range (- (:max (:year (key @state/date-atom))) min)
-        current-year (:value (:year (key @state/date-atom)))
-        last-year-update (:time-of-last-update (key @state/year-update))
+(defn update-year-info
+  [t left-right-key]
+  (let [min  (:min (:year (left-right-key @state/date-atom)))
+        range (- (:max (:year (left-right-key @state/date-atom))) min)
+        current-year (:value (:year (left-right-key @state/date-atom)))
+        last-year-update (:time-of-last-update (:year (left-right-key @state/year-update)))
         delta-year (int (- (* 5 t) last-year-update))]
     (when (> delta-year 0.5)
-      (swap! state/date-atom assoc-in [key :year :value] (+ min (rem (- (+ current-year delta-year) min) range)))
-      (swap! state/year-update assoc-in [key :time-of-last-update] (* 5 t)))))
+      (swap! state/date-atom assoc-in [left-right-key :year :value] (+ min (rem (- (+ current-year delta-year) min) range)))
+      (swap! state/year-update assoc-in [left-right-key :year :time-of-last-update] (* 5 t)))))
+
+(defn update-month-info
+  [t left-right-key]
+  (let [min  (:min (:month (left-right-key @state/date-atom)))
+        range (- (:max (:month (left-right-key @state/date-atom))) min)
+        current-year (:value (:month (left-right-key @state/date-atom)))
+        last-year-update (:time-of-last-update (:month (left-right-key @state/year-update)))
+        delta-month (int (- (* 1 t) last-year-update))]
+    (when (> delta-month 0.5)
+      (swap! state/date-atom assoc-in [left-right-key :month :value] (+ min (rem (- (+ current-year delta-month) min) range)))
+      (swap! state/year-update assoc-in [left-right-key :month :time-of-last-update] (* 1 t)))))
```

Comment:
Det bli mindre kod om du gör en funktion, tex update-year-month-info där year-month-key skickas in?